source/jormungandr/jormungandr/authentication.py

```diff
@@ -196,7 +229,14 @@ def get_all_available_instances(user):
 
     if not user:
         # for not-public navitia a user is mandatory
-        abort_request(user=user)
+        # To manage database error of the following type we should fetch one more time from database
+        # Can connect to database but at least one table/attribute is not accessible due to transaction problem
+        if can_read_user():
+            abort_request(user=user)
+
+        if not can_connect_to_database():
+            return None
```

Comment:
this piece of code is repeated a lot of times... Actually,  if you think about it, connectivity has become a preliminary condition for certain functions... you can make a decorator encapsulating that logic...

Reply:
A bit simplified